news_scraper/middlewares.py

```diff
@@ -50,8 +53,12 @@ def process_request(self, request, spider: Spider):
             spider.logger.info("Ignoring Request (already in output): %s", request.url)
             raise IgnoreRequest
 
-        # sample web proxy usage
-        # request.meta["proxy"] = "http://user:pass@brd.superproxy.io:22225"
+        # if spider.settings.getbool("USE_PROXY"):
+        #   request.meta["proxy"] = "http://user:pass@brd.superproxy.io:22225"
+        if spider.name == "moneycontrol" and spider.settings.getbool("USE_PROXY"):
+            proxy_url = os.getenv("PROXYURL")
+            request.meta["proxy"] = proxy_url
+            spider.logger.info("Using proxy")
```

Comment:
specify the proxy in the log as well for more clarity like `spider.logger.info(f"Using proxy: {proxy_url}")`

Reply:
ok I thought of not doing to maintain confidentiality